packages/plugins/response-cache-redis/test/response-redis-cache.spec.ts

```diff
@@ -0,0 +1,1155 @@
+              id: 2,
+              text: 'Comment 2 of User 2',
+            },
+          ],
+        },
+      ].slice(0, limit)
+    );
+
+    const schema = makeExecutableSchema({
+      typeDefs: /* GraphQL */ `
+        type Query {
+          users(limit: Int!): [User!]!
+        }
+
+        type User {
+          id: ID!
+          name: String!
+          comments: [Comment!]!
+          recentComment: Comment
+        }
+
+        type Comment {
+          id: ID!
+          text: String!
+        }
+      `,
+      resolvers: {
+        Query: {
+          users: spy,
+        },
+      },
+    });
+
+    const testInstance = createTestkit([useResponseCache({ cache })], schema);
+
+    const query = /* GraphQL */ `
+      query test($limit: Int!) {
+        users(limit: $limit) {
+          id
+          name
+          comments {
+            id
+            text
+          }
+        }
+      }
+    `;
+
+    // query and cache 2 users
+    await testInstance.execute(query, { limit: 2 });
+    // fetch 2 users from cache
+    await testInstance.execute(query, { limit: 2 });
+    // so just one query
+    expect(spy).toHaveBeenCalledTimes(1);
+
+    // we should have one response with operations
+    expect(await redis.keys('operations:*')).toHaveLength(1);
+
+    // query just one user
+    await testInstance.execute(query, { limit: 1 });
+
+    // since 2 users are in cache, we query again for the 1 as a response
+    expect(spy).toHaveBeenCalledTimes(2);
+  });
+
+  test('should purge response after it expired', async () => {
+    const spy = jest.fn(() => [
+      {
+        id: 1,
+        name: 'User 1',
+        comments: [
+          {
+            id: 1,
+            text: 'Comment 1 of User 1',
+          },
+        ],
+      },
+      {
+        id: 2,
+        name: 'User 2',
+        comments: [
+          {
+            id: 2,
+            text: 'Comment 2 of User 2',
+          },
+        ],
+      },
+    ]);
+
+    const schema = makeExecutableSchema({
+      typeDefs: /* GraphQL */ `
+        type Query {
+          users: [User!]!
+        }
+
+        type User {
+          id: ID!
+          name: String!
+          comments: [Comment!]!
+          recentComment: Comment
+        }
+
+        type Comment {
+          id: ID!
+          text: String!
+        }
+      `,
+      resolvers: {
+        Query: {
+          users: spy,
+        },
+      },
+    });
+
+    const testInstance = createTestkit([useResponseCache({ cache, ttl: 100 })], schema);
+
+    const query = /* GraphQL */ `
+      query test {
+        users {
+          id
+          name
+          comments {
+            id
+            text
+          }
+        }
+      }
+    `;
+
+    // query and cache
+    await testInstance.execute(query);
+    // from cache
+    await testInstance.execute(query);
+    // so queried just once
+    expect(spy).toHaveBeenCalledTimes(1);
+
+    await new Promise(resolve => setTimeout(resolve, 150));
```

Comment:
Could this be done by advancing fake timers instead?

Reply:
I used the same test that existed in the LRU cache-backed spec. But, I'll look at this. I think I saw fake timers used in another package.

Reply:
I think thought, that this test does want to really wait 150 msec so the item expires. But perhaps there is a better way to test that not using "real-time".